resources/metadata/whitesource.yaml

```diff
@@ -0,0 +1,523 @@
+        params:
+        - name: scanType
+          value: maven
+    - name: buildDescriptorFile
+      type: string
+      description: Explicit path to the build descriptor file.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType
+          value: mta
+    - name: buildDescriptorFile
+      type: string
+      description: Explicit path to the build descriptor file.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: ./package.json
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType
+          value: npm
+    - name: buildDescriptorFile
+      type: string
+      description: Explicit path to the build descriptor file.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: ./setup.py
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType
+          value: pip
+    - name: buildDescriptorFile
+      type: string
+      description: Explicit path to the build descriptor file.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: ./build.sbt
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType 
+          value: sbt
+    - name: buildDescriptorFile
+      type: string
+      description: Explicit path to the build descriptor file.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: ./dub.json
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType
+          value: dub
+    - name: createProductFromPipeline
+      type: bool
+      description: Whether to create the related WhiteSource product on the fly based on the supplied pipeline configuration.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: true
+    - name: securityVulnerabilities
+      type: bool
+      description: Whether security compliance is considered and reported as part of the assessment.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: true
+    - name: timeout
+      type: string
+      description: Timeout in seconds until a HTTP call is forcefully terminated.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: 0
+    - name: agentDownloadUrl
+      type: string
+      description: URL used to download the latest version of the WhiteSource Unified Agent.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: https://github.com/whitesource/unified-agent-distribution/releases/latest/download/${config.agentFileName}
```

Comment:
@nevskrem to my knowledge the *unified agent* is under heavy development, does it makes sense to freeze to a certain release?

Reply:
I think we should stick to `latest`.  For most cases we anyway cannot use unified agent so far for the other cases it makes sense that we learn with the latest version.
Teams can always freeze the version in case the consider this as helpful

Reply:
@OliverNocon Could you clarify what you mean by most cases we cannot use the unified agent? I thought the UA was required in order to conduct a scan (with the exception of using the Yarn/NPM plugin).

Reply:
there are issues for our current main use cases `maven` and `npm`.
So far we need to stick to the deprecated plugins in order to have consistent results.